monkey/monkey_island/cc/services/agent_plugin_service/flask_resources/agent_plugin_installation.py

```diff
@@ -0,0 +1,32 @@
+import logging
+from http import HTTPStatus
+
+from flask import make_response, request
+from flask_security import auth_token_required, roles_accepted
+
+from monkey_island.cc.flask_utils import AbstractResource
+from monkey_island.cc.services.authentication_service import AccountRole
+
+from .. import IAgentPluginService
+from ..errors import PluginInstallError
+
+logger = logging.getLogger(__name__)
+
+
+class AgentPluginInstallation(AbstractResource):
+    urls = ["/api/agent-plugins"]
+
+    def __init__(self, agent_plugin_service: IAgentPluginService):
+        self._agent_plugin_service = agent_plugin_service
+
+    @auth_token_required
+    @roles_accepted(AccountRole.ISLAND_INTERFACE.name)
```

Comment:
There's one ore more ETE tests that verify only the correct roles can access certain endpoints. We should update those tests to include this endpoint if possilbe.